### PR TITLE
refactor: replace lodash with small libraries

### PR DESCRIPTION
**Pull request by TrySound**
`camelcase` and `is-plain-obj` are small, single purpose and
dependency free utilities with builtin TS support. They cover all
left lodash usages.

### CODE REVIEW

**Comment by chimurai** [comment]
That's funny, just create almost the same branch; 9 minute prior to your PR :D

https://github.com/chimurai/http-proxy-middleware/commit/cadb25297dcc3a2ace236e9f46c7bbb652c604ac

File: src/path-rewriter.ts [hunk 1]
```diff
@@ -42,13 +42,9 @@ export function createPathRewriter(rewriteConfig) {
 function isValidRewriteConfig(rewriteConfig) {
   if (typeof rewriteConfig === 'function') {
     return true;
-  } else if (_.isPlainObject(rewriteConfig) && Object.keys(rewriteConfig).length !== 0) {
-    return true;
-  } else if (
-    rewriteConfig === undefined ||
-    rewriteConfig === null ||
-    _.isEqual(rewriteConfig, {})
```
Comment (chimurai) [inline src/path-rewriter.ts:50]: `_.isEqual(rewriteConfig, {})` check is now missing in the refactored code. Think the vanilla version would be: `JSON.stringify(rewriteConfig) === '{}'`.

Comment (TrySound) [reply src/path-rewriter.ts:50]: Comparing to empty object means it should be plain one, right? There is plain object check above. I moved Object.keys length check to return value instead.

**Comment by TrySound** [comment]
I would stick to Sindre's version. In my experience it's more popular than is-plain-object.

File: src/path-rewriter.ts (hunk 1 above)
Comment (chimurai) [reply src/path-rewriter.ts:50]: Ah, you're right. Missed that part. 👌

**Comment by chimurai** [comment]
Thanks!

**Pull request merged**